shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts

```diff
@@ -348,130 +348,18 @@ export class AppInsightsCore<CfgType extends IConfiguration = IConfiguration> im
-                        _initTimer = scheduleTimeout(() => {
-                            // set _isStatusSet to true
-                            // set active status
-                            // release queues
-                            _initTimer = null;
-                            if (!_isStatusSet) {
-                                _setStatus();
-                            }
-                           
-                        }, initTimeout);
-                   
-                        doAwaitResponse(allPromises, (response) => {
-                            try {
-                                if (_isStatusSet) {
-                                    // promises take too long to resolve, ignore them
-                                    // active status should be set by timeout already
-                                    return;
-                                }
-
-                                if (!response.rejected) {
-                                    let values = response.value;
-                                    if (values && values.length) {
-                                        // ikey
-                                        let ikeyRes = values[0];
-                                        _instrumentationKey = ikeyRes && ikeyRes.value;
-
-                                        // endpoint
-                                        if (values.length > 1) {
-                                            let endpointRes = values[1];
-                                            _endpoint = endpointRes &&  endpointRes.value;
-                                            
-                                        }
-
-                                    }
-                                    if (_instrumentationKey) {
-                                        // if ikey is null, no need to trigger extra dynamic changes for extensions
-                                        config.instrumentationKey = _instrumentationKey; // set config.instrumentationKey for extensions to consume
-                                        config.endpointUrl = _endpoint; // set config.endpointUrl for extensions to consume
-                                    }
-
-                                }
-
-                                // set _isStatusSet to true
-                                // set active status
-                                // release queues
-                                _setStatus();
-
-                            } catch (e) {
-                                if (!_isStatusSet){
-                                    _setStatus();
-                                }
-                            }
-
-                        });
-                    } else {
-                        // means no promises
-                        _setStatus();
-                       
-                    }
+                    
+                    _handleIKeyEndpointPromises(rootCfg);
```

Comment:
Also broke these out into smaller functions to contain the logic rather than just having this intline